ibis/backends/risingwave/__init__.py

```diff
@@ -268,3 +317,262 @@ def list_databases(
+            properties=sge.Properties(
+                expressions=sge.Properties.from_dict(connector_properties)
+            ),
+        )
+
+        create_stmt = create_stmt.sql(self.dialect) + data_and_encode_format(
+            data_format, encode_format, encode_properties
+        )
+
+        with self._safe_raw_sql(create_stmt):
+            pass
+
+        return self.table(name, database=database)
+
+    def drop_source(
+        self,
+        name: str,
+        *,
+        database: str | None = None,
+        force: bool = False,
+    ) -> None:
+        """Drop a Source.
+
+        Parameters
+        ----------
+        name
+            Source name to drop.
+        database
+            Name of the database where the view exists, if not the default.
+        force
+            If `False`, an exception is raised if the source does not exist.
+        """
+        src = sge.Drop(
+            this=sg.table(name, db=database, quoted=self.compiler.quoted),
+            kind="SOURCE",
+            exists=force,
+        )
+        with self._safe_raw_sql(src):
+            pass
+
+    def create_sink(
+        self,
+        name: str,
+        sink_from: str | None = None,
+        connector_properties: dict | None = None,
+        *,
+        obj: ir.Table | None = None,
+        database: str | None = None,
+        data_format: str | None = None,
+        encode_format: str | None = None,
+        encode_properties: dict | None = None,
+    ) -> None:
+        """Creating a sink.
+
+        Parameters
+        ----------
+        name
+            Sink name to Create.
+        sink_from
+            The table or materialized view name to sink from. Only one of `sink_from` or `obj` can be
+            provided.
+        connector_properties
+            The properties of the sink connector, providing the connector settings to push to the downstream data sink.
+            Refer https://docs.risingwave.com/docs/current/data-delivery/ for the required properties of different data sink.
+        obj
+            An Ibis table expression or pandas table that will be used to extract the schema and the data of the new table. Only one of `sink_from` or `obj` can be provided.
+        database
+            Name of the database where the source exists, if not the default.
+        data_format
+            The data format for the new source, e.g., "PLAIN". data_format and encode_format must be specified at the same time.
+        encode_format
+            The encode format for the new source, e.g., "JSON". data_format and encode_format must be specified at the same time.
+        encode_properties
+            The properties of encode format, providing information like schema registry url. Refer https://docs.risingwave.com/docs/current/sql-create-source/ for more details.
+        """
+        table = sg.table(name, db=database, quoted=self.compiler.quoted)
+
+        if encode_format is None != data_format is None:
```

Comment:
I think you want an `and` here -- as it's currently written I think this will error if both kwargs are `None`

```python
None is None != None is None
True != None is None
False is None
False
```